ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -146,10 +138,183 @@ const primalOutputStrings = {
+              return output.titanBlueWithPartner({ player: data.ShortName(partner) });
+            }
+          }
+        }
+
+        // From here on out, any response is for the current player.
+        if (matches.target !== data.me)
+          return response;
+
+        // Formless double tankbuster mechanic.
+        if (id === '00DA') {
+          if (data.role === 'tank')
+            return { alertText: output.formlessBusterAndSwap() };
+          // Not that you personally can do anything about it, but maybe this
+          // is your cue to yell on voice comms for cover.
+          return { alarmText: output.formlessBusterOnYOU() };
+        }
+
+        // Titan Mechanics (double blue handled above)
+        if (id === '00BB' && !data.seenFirstBombs)
+          return { alarmText: output.titanBlueSingular() };
+        if (id === '00B9')
+          return { alertText: output.titanYellowSpread() };
+        if (id === '00BA')
+          return { infoText: output.titanOrangeStack() };
+
+        // Statue laser mechanic.
+        const firstLaserMarker = '0091';
+        const lastLaserMarker = '0098';
+        if (id >= firstLaserMarker && id <= lastLaserMarker) {
+          // We could arguably tell you which giant you're tethered to by finding their position?
+          // And then saying something like "North" but that's probably more confusing than helpful.
+          // ids are sequential: #1 square, #2 square, #3 square, #4 square, #1 triangle etc
+          const decOffset = parseInt(id, 16) - parseInt(id, firstLaserMarker);
+          return {
+            alertText: [
+              output.square1(),
+              output.square2(),
+              output.square3(),
+              output.square4(),
+              output.triangle1(),
+              output.triangle2(),
+              output.triangle3(),
+              output.triangle4(),
+            ][decOffset],
+          };
+        }
+      },
+    },
+    {
+      id: 'E12S Promise Weight Cleanup',
+      netRegex: NetRegexes.startsUsing({ source: 'Eden\'s Promise', id: '58A5', capture: false }),
+      run: (data) => {
+        delete data.weightTargets;
+        data.seenFirstBombs = true;
+      },
+    },
+    {
+      id: 'E12S Promise Formless Judgment',
+      netRegex: NetRegexes.startsUsing({ source: 'Eden\'s Promise', id: '5858A9A5', capture: false }),
+      condition: Conditions.caresAboutPhysical(),
+      response: (data, _, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          formlessBusterAndSwap: {
+            en: 'Tank Buster + Swap',
+          },
+          tankBusters: {
+            en: 'Tank Busters',
+          },
+        };
+
+        // Already called out in the headmarker trigger.
+        if (data.formlessTargets && data.formlessTargets.includes(data.me))
+          return;
+
+        // TODO: should this call out who to cover if you are a paladin?
```

Comment:
By default, no. Maaaaaybe set it up as configurable like we've done for the TEA Wormhole bit or the Shiva whatever-it-was there?

Reply:
What's wrong for doing this with paladins? It's not a strat change option, it's more of a "how can I save the person who is second in enmity because the tank died and forgot to provoke".

Reply:
Oh, right, sorry! I was thinking of some "cover +invuln" nonsense like the Earth Shaker bit from O4s, but of course that doesn't apply here.